src/planner/plan_select.cpp

```diff
@@ -0,0 +1,67 @@
+#include <memory>
+#include <utility>
+
+#include "binder/bound_expression.h"
+#include "binder/bound_statement.h"
+#include "binder/bound_table_ref.h"
+#include "binder/statement/insert_statement.h"
+#include "binder/statement/select_statement.h"
+#include "binder/tokens.h"
+#include "common/enums/statement_type.h"
+#include "common/exception.h"
+#include "common/macros.h"
+#include "common/util/string_util.h"
+#include "execution/expressions/abstract_expression.h"
+#include "execution/expressions/column_value_expression.h"
+#include "execution/plans/abstract_plan.h"
+#include "execution/plans/filter_plan.h"
+#include "execution/plans/projection_plan.h"
+#include "fmt/format.h"
+#include "planner/planner.h"
+
+namespace bustub {
+
+auto Planner::PlanSelect(const SelectStatement &statement) -> std::unique_ptr<AbstractPlanNode> {
+  std::unique_ptr<AbstractPlanNode> plan = nullptr;
+
+  switch (statement.table_->type_) {
+    case TableReferenceType::EMPTY:
+      throw Exception("select value is not supported in planner yet");
+    default:
+      plan = PlanTableRef(*statement.table_);
+      break;
+  }
+
+  if (!statement.where_->IsInvalid()) {
```

Comment:
Curious question, how does it handle the case that the `where` clause is invalid?

Reply:
When users don't specify `where` clause, it will be invalid. So we don't handle it. e.g.,

```
select * from table;
```

In this case, `where` will be invalid in BoundSelect.

I agree that `invalid` might be confusing here. I'm thinking of changing it to `IsEmpty()`.